### PR TITLE
Wrong reference to Device Streaming Sample

### PR DESCRIPTION
**Pull request by Marusyk**
## Checklist
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [x] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [x] This pull-request is submitted against the `master` branch.

## Description of the changes
Small fix of link to sample with Device Streaming Sample

### CODE REVIEW

File: iothub/service/samples/readme.md [hunk 1]
```diff
@@ -17,11 +17,11 @@ Service samples were moved to [Azure-Samples/azure-iot-samples-csharp][samples-r
 [samples-repo]: https://github.com/Azure-Samples/azure-iot-samples-csharp
 [service-samples]: https://github.com/Azure-Samples/azure-iot-samples-csharp/tree/master/iot-hub/Samples/service
 [adm-sample]: https://github.com/Azure-Samples/azure-iot-samples-csharp/tree/master/iot-hub/Samples/service/AutomaticDeviceManagementSample
-[device-streaming-sample]: https://github.com/Azure-Samples/azure-iot-samples-csharp/tree/master/iot-hub/Samples/service/DeviceStreamingSample
```
Comment (abhipsaMisra) [inline iothub/service/samples/readme.md:20]: Thanks for catching this dead link!

We've actually moved our "preview" samples to the `preview` branch, so this should point to the following instead: https://github.com/Azure-Samples/azure-iot-samples-csharp/tree/preview/iot-hub/Samples/service/DeviceStreamingSample

File: iothub/service/samples/readme.md [hunk 2]
```diff
@@ -17,11 +17,11 @@ Service samples were moved to [Azure-Samples/azure-iot-samples-csharp][samples-r
...
+[device-streaming-sample]: https://github.com/Azure-Samples/azure-iot-samples-csharp/tree/master/iot-hub/Quickstarts/device-streams-proxy
```
Comment (drwill-ms) [inline iothub/service/samples/readme.md:20]: The former one was broken, but the true URL is <https://github.com/Azure-Samples/azure-iot-samples-csharp/tree/preview/iot-hub/Samples/service/DeviceStreamingSample>

```suggestion
[device-streaming-sample]: https://github.com/Azure-Samples/azure-iot-samples-csharp/tree/preview/iot-hub/Samples/service/DeviceStreamingSample
```

**Review by drwill-ms — APPROVED** [review]

**Comment by drwill-ms** [comment]
Thanks @Marusyk !

**Pull request merged**